sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLCLIDriver.scala

```diff
@@ -74,7 +74,16 @@ private[hive] object SparkSQLCLIDriver {
       System.exit(1)
     }
 
-    val sessionState = new CliSessionState(new HiveConf(classOf[SessionState]))
+    val localMetastore = {
+      val temp = Utils.createTempDir()
+      temp.delete()
```

Comment:
Instead of this I'd use "$temp/something" as the metastore directory, to avoid the "temp" directory being created by another process before Hive has a chance to do it.